src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext, useState } from 'react';
+import { TodosContext } from '../../TodosContext';
+import { TodoItem } from '../TodoItem';
+import { TodosFilter } from '../TodosFilter';
+
+export const TodoList = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+
+  const [filterParam, setFilterParam] = useState('All');
+
+  const handleToggleAll = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const todosCopy = [...todos].map(todo => ({
+      ...todo, completed: event.target.checked,
+    }));
```

Comment:
map creates a new array, you don't need to create a copy.
Also, use functional `setState`